### PR TITLE
868804 - Prepare UG for Neumorphic UI chart type in .NET MAUI column chart

### PR DESCRIPTION
**Pull request by DhanarajRajendran**
### Description ###

Added the Document for the Neumorphic UI column chart.

### Output ###

Screenshot 2024-02-16 123614

### CODE REVIEW

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+title: Neumorphic UI Column Chart in .NET MAUI Chart control | Syncfusion
+description: Learn here about how to create the Neumorphic UI chart types using Syncfusion .NET MAUI Chart (SfCartesianChart) control.
+platform: maui
+control: SfCartesianChart
+documentation: ug
+---
+
+# Neumorphic UI Chart using Syncfusion .NET MAUI Column Chart
+
+Learn to create visually stunning Neumorphic charts in using Syncfusion [.NET MAUI Column Chart](). Follow a step-by-step tutorial utilizing a custom class that inherits from `IDrawable` and `Graphics view` for a seamless blend of design and data representation in your application.
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:12]: Update UG link for column chart

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
+---
+layout: post
+title: Neumorphic UI Column Chart in .NET MAUI Chart control | Syncfusion
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:3]: @DhanarajRajendran 
Modify the title as Neumorphic UI Chart in .NET MAUI Chart control

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+description: Learn here about how to create the Neumorphic UI chart types using Syncfusion .NET MAUI Chart (SfCartesianChart) control.
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:4]: @DhanarajRajendran 
In description, mention as -> Neumorphic UI column chart

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md (hunk 1 above)
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:12]: @DhanarajRajendran 
Modify the first line as 
  Learn to create visually stunning Neumorphism UI charts using Syncfusion [.NET MAUI Column Chart]

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+Learn to create visually stunning Neumorphic charts in using Syncfusion [.NET MAUI Column Chart](). Follow a step-by-step tutorial utilizing a custom class that inherits from `IDrawable` and `Graphics view` for a seamless blend of design and data representation in your application.
+
+
+## Step 1: Implementing SfShadowDrawer Class
+
+The `SfShadowDrawer` class is a base class that implements the `IDrawable` interface. It serves as a foundation for creating drawable elements with shadow effects. This class provides properties and methods to control shadow properties, such as offset, blur, color, and opacity.
+
+## Step 2: Implementing Draw Method
+
+Implements the `IDrawable` interface method `Draw`. This method is responsible for drawing the shadow by applying the shadow effect and filling the rounded rectangle.
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:21]: @DhanarajRajendran 
Looks like sentence not having proper information & incomplete, modify as
Implement the Draw method for the IDrawable interface in the custom SfShadowDrawer class.

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+        canvas.ClipPath(GetDrawPath(dirtyRect, (float)radius));
+        DrawShadow(canvas, dirtyRect);
+        canvas.RestoreState();
+    }
+
+{% endhighlight SfShadowDrawer %}
+
+{% endtabs %}
+
+## Step 3: Implement a Method to Apply shadow
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:41]: @DhanarajRajendran 
Combine step 2 & 3, add title as apply shadow like that

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+        canvas.RestoreState();
+    }
+
+{% endhighlight SfShadowDrawer %}
+
+{% endtabs %}
+ 
+## Step 4: Implementing SfNeumorphismDrawer Class
+
+The `SfNeumorphismDrawer` class is a custom drawable class that extends `SfShadowDrawer` and implements the `IDrawable` interface. It is specifically designed for creating a neumorphic (soft shadow) effect in user interfaces. This class provides properties and methods to control various aspects of the neumorphic shadow, such as color, offset, and opacity.
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:64]: @DhanarajRajendran 
In SfNeumorphismDrawer class, you mentioned as -> implements the `IDrawable` interface, remove it

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+
+It overrides the `DrawShadow` method from the base class `SfShadowDrawer` to implement the neumorphic shadow drawing logic. The shadow is drawn based on the pressed state, adjusting the offset and shadow positions accordingly.
+
+{% tabs %}
+
+{% highlight SfNeumorphismDrawer %}
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:70]: @DhanarajRajendran 
check tab

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 8]
```diff
@@ -0,0 +1,245 @@
...
+
+        graphicsView = new GraphicsView();
+        graphicsView.Margin = new Thickness(0);
+        graphicsView.BackgroundColor = Colors.Transparent;
+        graphicsView.SetBinding(GraphicsView.DrawableProperty, new Binding() { Path = nameof(Drawable), Source = this });
+
+        grid.Children.Add(graphicsView);
+        base.Content = grid;
+    }
+    public void Invalidate()
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:128]: @DhanarajRajendran 
Is this method is required?

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+
+{% endhighlight SfNeumorphismView %}
+
+{% endtabs %}
+
+## Step 6: Implementing the SfNeumorphismColumnSeries and SfNeumorphismColumnSegment Classes
+
+The `SfNeumorphismColumnSeries` class is a custom implementation of the `ColumnSeries` in the Syncfusion.Maui.Charts library. It introduces a neumorphic (soft shadow) effect to column series by associating a `SfNeumorphismDrawer` with each column segment.
+
+This class overrides the base method to create a custom `SfNeumorphismColumnSegment` instead of the default `ColumnSegment`. Associates the `Drawable` property with the newly created segment.
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:141]: @DhanarajRajendran 
Include code snippet

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+        return new SfNeumorphismColumnSegment(Drawable);
+    }
+    }
+
+{% endhighlight SfNeumorphismColumnSeries %}
+
+{% endtabs %}
+
+
+The `SfNeumorphismColumnSegment` class is a custom implementation of the `ColumnSegment` in the Syncfusion.Maui.Charts library. It enhances the default column segment by incorporating a neumorphic shadow effect using the associated `SfNeumorphismDrawer`.
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:160]: @DhanarajRajendran 
You have explained SfNeumorphismColumnSegment previously, why need to explain again?

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+    {
+        if (Series is ColumnSeries series && series.ActualYAxis is NumericalAxis yAxis)
+        {
+            var top = yAxis.ValueToPoint(Convert.ToDouble(yAxis.Maximum ?? double.NaN));
+    
+            var trackRect = new RectF() { Left = Left, Top = top, Right = Right, Bottom = Bottom };
+    
+            Drawable.Draw(canvas, trackRect);
+        }
+    }
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:180]: @DhanarajRajendran 
Code alignment is not proper

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+                                IsPressedState="True" Opacity="0.3" 
+                                LightOpacity="1" CornerRadius="20" 
+                                BackgroundColor="#F5F5F5" x:Key="pressedDrawable" />
+    </ContentPage.Resources>
+
+{% endhighlight MainPage.xaml}
+
+{% endtabs %}
+
+## Step 8: Applying the styles.
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:213]: @DhanarajRajendran 
Combine 7& 8 step

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 13]
```diff
@@ -0,0 +1,245 @@
...
+        </chart:SfCartesianChart>
+    </control:SfNeumorphismView>
+
+{% endhighlight MainPage.xaml}
+
+{% endtabs %}
+
+Here is the final output of the Syncfusion .NET MAUI Neumorphic Column Chart.
+
+![Final Output of Syncfusion .NET MAUI Neumorphic Column Chart](Neumorphic_Chart_Image/maui-neumorphic_ui_chart_image.png)
```
Comment (VimalaThirumalaiKumarSF3739) [inline MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:245]: @DhanarajRajendran 
Add sample also

File: maui-toc.html [hunk 14]
```diff
@@ -310,6 +310,7 @@
                     <li><a href="/maui/cartesian-charts/errorbar-chart">Error Bar Chart</a></li>
                     <li><a href="/maui/cartesian-charts/OHLC">OHLC Chart</a></li>
                     <li><a href="/maui/cartesian-charts/candle">Candle Chart</a></li>
+                    <li><a href="/maui/cartesian-charts/neumorphicUI-column-chart">Neumorphic Column Chart</a></li>
```
Comment (VimalaThirumalaiKumarSF3739) [inline maui-toc.html:313]: @DhanarajRajendran 
Modify section title as Neumorphic UI Chart

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md (hunk 9 above)
Comment (DhanarajRajendran) [reply MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:141]: Added the code snippet for both series and segment, below their contents.

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md (hunk 10 above)
Comment (DhanarajRajendran) [reply MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:160]: I haven't explained it before.

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md (hunk 3 above)
Comment (DhanarajRajendran) [reply MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:4]: Addressed the comments.

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md (hunk 1 above)
Comment (DhanarajRajendran) [reply MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:12]: Addressed the comments.

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md [hunk 15]
```diff
@@ -0,0 +1,245 @@
...
+
+The `SfShadowDrawer` class is a base class that implements the `IDrawable` interface. It serves as a foundation for creating drawable elements with shadow effects. This class provides properties and methods to control shadow properties, such as offset, blur, color, and opacity.
+
+## Step 2: Implementing Draw Method
+
+Implements the `IDrawable` interface method `Draw`. This method is responsible for drawing the shadow by applying the shadow effect and filling the rounded rectangle.
+
+{% tabs %}
+
+{% highlight SfShadowDrawer %}
```
[the remark this answers is not in the document]
Comment (DhanarajRajendran) [reply MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:25]: I have used the class name, to differentiate the codes with class.

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md (hunk 5 above)
Comment (DhanarajRajendran) [reply MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:41]: Addressed the comments.

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md (hunk 8 above)
Comment (DhanarajRajendran) [reply MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:128]: Not required for the documentation, So removed that content.

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md (hunk 11 above)
Comment (DhanarajRajendran) [reply MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:180]: Aligned the code properly.

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md (hunk 13 above)
Comment (DhanarajRajendran) [reply MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:245]: Need to provide the link.

File: MAUI/Cartesian-Charts/neumorphicUI-column-chart.md (hunk 12 above)
Comment (DhanarajRajendran) [reply MAUI/Cartesian-Charts/neumorphicUI-column-chart.md:213]: Addressed the comments.

File: maui-toc.html (hunk 14 above)
Comment (DhanarajRajendran) [reply maui-toc.html:313]: Addressed the comments.

**Comment by VimalaThirumalaiKumarSF3739** [comment]
As discussed with PO, published [KB](https://support.syncfusion.com/kb/article/15491/how-to-create-the-neumorphic-ui-chart-using-net-maui-chart-sfcartesianchart) for it. So, closing the PR

**Pull request closed without merging**